ksqldb-serde/src/main/java/io/confluent/ksql/serde/kafka/KafkaFormat.java

```diff
@@ -59,4 +62,11 @@ public String name() {
 
     return KafkaSerdeFactory.createSerde(schema);
   }
+
+  @Override
+  public boolean supportKeyType(SqlType type) {
+    return type instanceof SqlPrimitiveType
```

Comment:
Is this first piece necessary?

Reply:
I think so, just a double safety approach.